src/org/purefn/river/batch.clj

```diff
@@ -13,24 +13,29 @@
            [org.purefn.river Message]))
 
 (defn kafka-consumer
-  [{:keys [::group-id ::bootstrap-servers ::deserializer]}]
-  (if deserializer 
-    (KafkaConsumer.
-     {"auto.offset.reset" "earliest"
-      "bootstrap.servers" bootstrap-servers
-      "enable.auto.commit" false
-      "group.id" group-id
-      "client.id" group-id
-      "key.deserializer" (::key.deserializer deserializer)
-      "value.deserializer" (::value.deserializer deserializer)})
-    (KafkaConsumer.
-     {"auto.offset.reset" "earliest"
-      "bootstrap.servers" bootstrap-servers
-      "enable.auto.commit" false
-      "group.id" group-id
-      "client.id" group-id}
-     (serdes/nippy-deserializer)
-     (serdes/nippy-deserializer))))
+  [{:keys [::group-id ::bootstrap-servers ::deserializer
```

Comment:
Have the standards on docstrings for public functions dropped so far since I left? ;)

Reply:
It is technically public, but not sure I'd consider it that way since the client wouldn't directly interact with this, they just supply the config (which is spec'd and validated). but its a fair point and I have added a doc string.

Reply:
Ah, I see now. The public function is `batch-consumer`. (It also doesn't have a docstring)

Reply:
I moved the config details to batch consumer and added some more bits.